src/sentry/coreapi.py

```diff
@@ -335,10 +343,10 @@ def _process_data_timestamp(self, data, current_datetime=None):
             current_datetime = datetime.now()
 
         if value > current_datetime + timedelta(minutes=1):
-            raise InvalidTimestamp('Invalid value for timestamp (in future): %r' % value)
+            raise FutureInvalidTimestamp('%s' % (value - current_datetime))
 
         if value < current_datetime - timedelta(days=30):
-            raise InvalidTimestamp('Invalid value for timestamp (too old): %r' % value)
+            raise PastInvalidTimestamp('%s' % (value - current_datetime))
```

Comment:
Another option would be to port this logic that we use to show reporting latency, though I think you might want a more precise value here: https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/views/groupDetails/eventToolbar.jsx#L12-L31